### DISCUSSION TITLE
VRoid SDK with Unity 2020.3.9f1 or later

### DISCUSSION THREAD

**Discussion by AnriEU** [pixiv/vroid-sdk-developers#5] [General]
Has this problem been fixed?

**Comment by Haresaku** [comment]
When using Unity2020, Unity 2020.3.16f1 or later version is recommended.

The VRoid SDK uses UnityWebRequest, but UnityWebRequest on Windows does not support gzip, so the VRoid SDK manually decompresses the gzip-compressed contents when it receives them internally.

This workaround was not a problem before Unity 2020, but since Unity 2020, it becomes an error just by receiving Content-Encoding in the response, so this workaround can no longer be used.

Unity 2020.3.16f1 officially supports gzip on Windows platforms, so please use that version or later.

https://issuetracker.unity3d.com/issues/unitywebrequest-fails-on-windows-if-gzip-compression-is-used
https://unity3d.com/jp/unity/whats-new/2020.3.16

**Comment by simplestargame** [comment]
Unity2021.2.0f1 よりまたこちらの問題が発生することを確認しました。
次のエラーがコンソールに出力されます
Curl error 61: Unrecognized content encoding type. libcurl understands identity content encodings.
2020.3.9～2020.3.15 までの時のように、しばらく未対応で、その後対応が入ることを期待してます。
ですので、しばらくは 2020.3.21 を使い続けます。

やはり VRoidSDK 側もこちらのエラーを解決する手段を提供いただけると今後うれしいと思います

**Reply by Haresaku** [reply]
ご報告ありがとうございます。こちらの問題について、次回のバージョンで動作できるように対応を行う予定です。
お手数ですが、2020.3.21をご利用ください

**Reply by simplestargame** [reply]
こちら Unity 2021.2.4 より
>Networking: UnityWebRequest now supports gzip compression on Windows. 
https://unity3d.com/pt/unity/whats-new/2021.2.4

解決していることを確認できました。上記報告のエラーなく Unity 2021.2.4  にて VRoidSDK を利用できています。
